ibis-server/app/routers/v2/connector.py

```diff
@@ -44,16 +50,47 @@ async def query(
     with tracer.start_as_current_span(
         name=span_name, kind=trace.SpanKind.SERVER, context=build_context(headers)
     ):
-        rewritten_sql = await Rewriter(
-            dto.manifest_str,
-            data_source=data_source,
-            java_engine_connector=java_engine_connector,
-        ).rewrite(dto.sql)
-        connector = Connector(data_source, dto.connection_info)
-        if dry_run:
-            connector.dry_run(rewritten_sql)
-            return Response(status_code=204)
-        return ORJSONResponse(to_json(connector.query(rewritten_sql, limit=limit)))
+        cached_result = None
+        cache_hit = False
+        enable_cache = dto.enable_cache
+
+        if enable_cache:
+            cached_result = query_cache_manager.get(
+                str(data_source), dto.sql, dto.connection_info
+            )
+            cache_hit = cached_result is not None
+
+        # Cache Hit !
+        if cached_result is not None:
+            response = ORJSONResponse(to_json(cached_result))
+            response.headers["X-Cache-Hit"] = str(cache_hit).lower()
```

Comment:
Instead of a header field, how about using the query parameter? Just like dry-run, the client can invoke this API like
```
post .../v2/bigquery/query?cahceHit=true
```